src/objects/OrientationGuide.ts

```diff
@@ -0,0 +1,145 @@
+        const glow = this.scene.add.circle(
+            0, 0, 110, 0xffffff, 0.2
+        ).setOrigin(0.5);
+
+        // Phone rotation image
+        const rotateImage = this.scene.add.image(
+            0, 0, 'rotate-phone'
+        ).setScale(0.3);
+        
+        // Add to rotation container
+        rotateContainer.add([glow, rotateImage]);
+
+        // Instruction text
+        const instructionText = this.scene.add.text(
+            CONFIG.GAME_WIDTH / 2,
+            CONFIG.GAME_HEIGHT / 2 + 80,
+            'Please rotate your device\nfor best experience',
+            {
+                fontFamily: 'PixelGame',
+                fontSize: '20px',
+                color: '#ffffff',
+                stroke: '#000000',
+                strokeThickness: 4,
+                align: 'center'
+            }
+        ).setOrigin(0.5);
+
+        // "Tap to continue" text
+        const tapText = this.scene.add.text(
+            CONFIG.GAME_WIDTH / 2,
+            CONFIG.GAME_HEIGHT / 2 + 150,
+            'Tap to Continue',
+            {
+                fontFamily: 'PixelGame',
+                fontSize: '20px',
+                color: '#ffffff',
+                stroke: '#000000',
+                strokeThickness: 4
+            }
+        ).setOrigin(0.5);
+
+        this.overlay.add([bg, rotateContainer, instructionText, tapText]);
+        this.overlay.setDepth(1000);
+        this.overlay.setVisible(false);
+
+        // Make the entire overlay interactive
+        bg.setInteractive();
+        bg.on('pointerdown', this.hideOverlay, this);
+        
+        // Add pulse effect to the rotate container
+        this.pulseEffect = this.scene.tweens.add({
+            targets: rotateContainer,
+            scaleX: 1.1,
+            scaleY: 1.1,
+            duration: 800,
+            yoyo: true,
+            repeat: -1,
+            ease: 'Sine.easeInOut',
+            paused: true
+        });
+    }
+
+    private checkOrientation(): void {
+        // Use matchMedia to check if we're in portrait mode on a mobile device
+        // This is the modern best practice for detecting orientation
+        const mediaQuery = window.matchMedia("(orientation: portrait) and (max-width: 600px)");
+        
+        // Show overlay if in portrait mode
+        if (mediaQuery.matches) {
+            this.showOverlay();
+            OrientationGuide.hasBeenShownBefore = true;
+            
+            // Add event listener to hide overlay when device is rotated
+            const orientationChangeHandler = () => {
+                if (!mediaQuery.matches) {
+                    this.hideOverlay();
+                    // Remove the event listener after hiding
+                    mediaQuery.removeEventListener('change', orientationChangeHandler);
+                }
+            };
+            
+            // Listen for orientation changes
+            mediaQuery.addEventListener('change', orientationChangeHandler);
```

Comment:
so 3rd parameter of addEventListener, you can do once: and not worry about having to remove it or track if its been seen before, etc

```suggestion
            mediaQuery.addEventListener('change', orientationChangeHandler, { once: true });
```

At least if mediaQuery.addEventListener works like it does for document.addEventListener('